src/rest.cpp

```diff
@@ -252,6 +252,60 @@ static bool rest_chaininfo(AcceptedConnection* conn,
     return true; // continue to process further HTTP reqs on this cxn
 }
 
+static bool rest_mempool_info(AcceptedConnection* conn,
+                                   const std::string& strReq,
+                                   const std::map<std::string, std::string>& mapHeaders,
+                                   bool fRun)
+{
+    vector<string> params;
+    const RetFormat rf = ParseDataFormat(params, strReq);
+
+    switch (rf) {
+    case RF_JSON: {
+        Array rpcParams;
+        Value mempoolInfoObject = getmempoolinfo(rpcParams, false);
+
+        string strJSON = write_string(mempoolInfoObject, false) + "\n";
+        conn->stream() << HTTPReply(HTTP_OK, strJSON, fRun) << std::flush;
+        return true;
+    }
+    default: {
+        throw RESTERR(HTTP_NOT_FOUND, "output format not found (available: json)");
+    }
+    }
+
+    // not reached
+    return true; // continue to process further HTTP reqs on this cxn
+}
+
+static bool rest_mempool_contents(AcceptedConnection* conn,
+                                   const std::string& strReq,
+                                   const std::map<std::string, std::string>& mapHeaders,
+                                   bool fRun)
+{
+    vector<string> params;
+    const RetFormat rf = ParseDataFormat(params, strReq);
+
+    switch (rf) {
+    case RF_JSON: {
+        Array rpcParams;
+	rpcParams.push_back(true);
+
+        Value mempoolObject = getrawmempool(rpcParams, false);
```

Comment:
I'm not sure if we should just loop through RPC commands. I know that i did the same when i added the `/chaininfos` REST command. This will tie the REST interface to the current RPC implementation. But duplicating code is also something we should avoid.
I don't have a solution here i just wanted to bring up this concern.